src/main/java/org/folio/inventory/DataImportConsumerVerticle.java

```diff
@@ -97,87 +74,29 @@ public class DataImportConsumerVerticle extends AbstractVerticle {
-    ConsortiumDataCache consortiumDataCache = new ConsortiumDataCache(vertx, client);
-    MappingMetadataCache mappingMetadataCache = MappingMetadataCache.getInstance(vertx, client);
-    DataImportKafkaHandler dataImportKafkaHandler = new DataImportKafkaHandler(
-      vertx, storage, client, profileSnapshotCache, kafkaConfig, mappingMetadataCache, consortiumDataCache);
+    var dataImportKafkaHandler = new DataImportKafkaHandler(vertx, getStorage(), getHttpClient(), getProfileSnapshotCache(),
+      getKafkaConfig(), getMappingMetadataCache(), consortiumDataCache);
 
-    List<Future<KafkaConsumerWrapper<String, String>>> futures = EVENT_TYPES.stream()
-      .map(eventType -> createKafkaConsumerWrapper(kafkaConfig, eventType, dataImportKafkaHandler))
-      .collect(Collectors.toList());
+    var futures = EVENT_TYPES.stream()
+      .map(type -> super.createConsumer(type.value()))
+      .map(consumerWrapper -> consumerWrapper.start(dataImportKafkaHandler, ConsumerWrapperUtil.constructModuleName())
+        .map(consumerWrapper)
+      )
+      .toList();
 
     GenericCompositeFuture.all(futures)
       .onFailure(startPromise::fail)
-      .onSuccess(ar -> {
-        futures.forEach(future -> consumerWrappers.add(future.result()));
-        startPromise.complete();
-      });
+      .onSuccess(ar -> startPromise.complete());
   }
 
   @Override
-  public void stop(Promise<Void> stopPromise) {
-    List<Future<Void>> stopFutures = consumerWrappers.stream()
-      .map(KafkaConsumerWrapper::stop)
-      .collect(Collectors.toList());
-
-    GenericCompositeFuture.join(stopFutures).onComplete(ar -> stopPromise.complete());
-  }
-
-  private Future<KafkaConsumerWrapper<String, String>> createKafkaConsumerWrapper(KafkaConfig kafkaConfig, DataImportEventTypes eventType,
-                                                                                  AsyncRecordHandler<String, String> recordHandler) {
-    SubscriptionDefinition subscriptionDefinition = KafkaTopicNameHelper.createSubscriptionDefinition(kafkaConfig.getEnvId(),
-      KafkaTopicNameHelper.getDefaultNameSpace(), eventType.value());
-
-    KafkaConsumerWrapper<String, String> consumerWrapper = KafkaConsumerWrapper.<String, String>builder()
-      .context(context)
-      .vertx(vertx)
-      .kafkaConfig(kafkaConfig)
-      .loadLimit(loadLimit)
-      .globalLoadSensor(new GlobalLoadSensor())
-      .subscriptionDefinition(subscriptionDefinition)
-      .build();
-
-    return consumerWrapper.start(recordHandler, ConsumerWrapperUtil.constructModuleName())
-      .map(consumerWrapper);
+  protected Logger getLogger() {
+    return LOGGER;
   }
 
-  private int getLoadLimit() {
-    return Integer.parseInt(System.getProperty("inventory.kafka.DataImportConsumer.loadLimit", "5"));
-  }
-
-  private int getMaxDistributionNumber() {
-    return Integer.parseInt(System.getProperty("inventory.kafka.DataImportConsumerVerticle.maxDistributionNumber", "100"));
```

Comment:
The `getMaxDistributionNumber` method in `KafkaConsumerVerticle` will not work here :( because the system property has "Verticle" in it. In `getConsumerProperty`, we remove "Verticle" :(